java/com/google/cose/structures/EncryptStructure.java

```diff
@@ -0,0 +1,31 @@
+package com.google.cose.structures;
+
+import co.nstant.in.cbor.CborBuilder;
+import co.nstant.in.cbor.builder.ArrayBuilder;
+import co.nstant.in.cbor.model.ByteString;
+import co.nstant.in.cbor.model.DataItem;
+import co.nstant.in.cbor.model.Map;
+import com.google.cose.utils.CborUtils;
+
+/**
+ * Encodes the Enc_Structure as mentioned in COSE RFC section 5.3
+ */
+public class EncryptStructure {
+  private final String context;
+  private final Map protectedHeaders;
+  private final byte[] externalAad;
+
+  public EncryptStructure(String context, Map headers, byte[] externalAad) {
+    this.context = context;
```

Comment:
Can you make context an enumeration? IIUC, there's a very limited set of allowed values, and making an enumeration will help prevent silly mistakes like typos.

Reply:
The same comment applies to all the structure types. I think their context values are all very limited.

Reply:
Converted in all structure types.